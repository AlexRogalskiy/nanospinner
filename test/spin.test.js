import { createSpinner } from '../index.js'

let stdout = { out: '' }
stdout.write = symbols => {
  stdout.out += symbols
}

it('spins default frames', () => {
  let spinner = createSpinner('', { stream: stdout })

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.spin()

  expect(stdout.out).toMatchSnapshot()
})
