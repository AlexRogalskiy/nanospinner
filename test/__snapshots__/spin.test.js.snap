// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spins default frames 1`] = `"[1G[33m‚†ã[39m [1G[33m‚†ô[39m [1G[33m‚†π[39m [1G[33m‚†∏[39m [1G[33m‚†º[39m [1G[33m‚†¥[39m [1G[33m‚†¶[39m [1G[33m‚†ß[39m [1G[33m‚†á[39m [1G[33m‚†è[39m "`;
